Source/OpalGraphics/CGPath.m

```diff
@@ -311,9 +592,87 @@ void CGPathAddArcToPoint(
   CGFloat y1,
   CGFloat x2,
   CGFloat y2,
-  CGFloat r)
+  CGFloat radius)
 {
-  // FIXME:
+  CGFloat dx1, dy1, dx2, dy2;
+  CGFloat l, a1, a2;
+  CGPoint p;
+
+  p = CGPathGetCurrentPoint(path);
+ 
+  dx1 = p.x - x1;
+  dy1 = p.y - y1;
+  
+  l= dx1*dx1 + dy1*dy1;
+  if (l <= 0)

```

Comment:
Shpuld we use a shared bit of logic between this function and CGPathAddArc when to lineTo, moveTo or do nothing? Both current versions look suboptimal.
I have to admit, I did not look at the full version of the function and may be totally off here. What I am think of is that when ever we start an arc we first must decide how to get to the first point of it. And this should be properly implemented.